From 753e2c1340d24a31979429c71ebc5e8ae3e3a2c5 Mon Sep 17 00:00:00 2001
From: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
Date: Thu, 6 Aug 2026 22:50:16 -0300
Subject: [PATCH] dt-bindings: mfd: add EcoNet EN751221 GPIO syscon

Document the syscon and simple-MFD wrapper around the GPIO
register block shared by the EN7512, EN7513, EN7521 and EN7526
SoC families.

Signed-off-by: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
---
 .../mfd/econet,en751221-gpio-sysctl.yaml      | 57 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 58 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/mfd/econet,en751221-gpio-sysctl.yaml

diff --git a/Documentation/devicetree/bindings/mfd/econet,en751221-gpio-sysctl.yaml b/Documentation/devicetree/bindings/mfd/econet,en751221-gpio-sysctl.yaml
new file mode 100644
index 000000000000..e0a873e18d79
--- /dev/null
+++ b/Documentation/devicetree/bindings/mfd/econet,en751221-gpio-sysctl.yaml
@@ -0,0 +1,57 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/mfd/econet,en751221-gpio-sysctl.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: EcoNet EN7512/EN7521 GPIO System Controller
+
+maintainers:
+  - Matheus Sampaio Queiroga <srherobrine20@gmail.com>
+
+description:
+  GPIO register block and MFD container used by the EcoNet EN7512, EN7513,
+  EN7521 and EN7526 SoC families.
+
+properties:
+  compatible:
+    items:
+      - const: econet,en751221-gpio-sysctl
+      - const: syscon
+      - const: simple-mfd
+
+  reg:
+    maxItems: 1
+
+  pinctrl:
+    type: object
+    $ref: /schemas/pinctrl/econet,en751221-pinctrl.yaml
+
+required:
+  - compatible
+  - reg
+  - pinctrl
+
+additionalProperties: false
+
+examples:
+  - |
+    system-controller@1fbf0200 {
+      compatible = "econet,en751221-gpio-sysctl", "syscon", "simple-mfd";
+      reg = <0x1fbf0200 0x7c>;
+
+      pinctrl: pinctrl {
+        compatible = "econet,en751221-pinctrl";
+        airoha,chip-scu = <&chip_scu>;
+
+        interrupt-parent = <&intc>;
+        interrupts = <10>;
+
+        gpio-controller;
+        #gpio-cells = <2>;
+        gpio-ranges = <&pinctrl 0 0 64>;
+
+        interrupt-controller;
+        #interrupt-cells = <2>;
+      };
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index 851df88bf9bc..b5d67b6c1d10 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -20332,6 +20332,7 @@ PIN CONTROLLER - AIROHA
 M:	Lorenzo Bianconi <lorenzo@kernel.org>
 L:	linux-mediatek@lists.infradead.org (moderated for non-subscribers)
 S:	Maintained
+F:	Documentation/devicetree/bindings/mfd/econet,en751221-gpio-sysctl.yaml
 F:	Documentation/devicetree/bindings/pinctrl/airoha,en7581-pinctrl.yaml
 F:	drivers/pinctrl/airoha/pinctrl-airoha.c
 
-- 
2.55.0

